x-pack/packages/kbn-entities-schema/src/schema/common.ts

```diff
@@ -141,9 +141,10 @@ export type MetadataField = z.infer<typeof metadataSchema>;
 export const identityFieldsSchema = z
   .object({
     field: z.string(),
-    optional: z.boolean(),
+    optional: z.boolean().optional().default(false),
+    fallback: z.string().optional(),
```

Comment:
can we already pass the default value in the schema so we don't have to deal with conditional in code ?

Reply:
Optional fields were removed!